luaui/Widgets/Shaders/DrawPrimitiveAtUnit.geom.glsl

```diff
@@ -82,55 +82,60 @@ void main(){
-			offsetVertex4( width * 0.5, 0.0,  length * 0.5, 0.0, 1.0);
-			offsetVertex4( width * 0.5, 0.0, -length * 0.5, 0.0, 0.0);
-			offsetVertex4(-width * 0.5, 0.0,  length * 0.5, 1.0, 1.0);
-			offsetVertex4(-width * 0.5, 0.0, -length * 0.5, 1.0, 0.0);
+		if (numVertices == 4u){ // A quad
+			offsetVertex4( width * 0.5, 0.0,  length * 0.5, 0.0, 1.0, 1.414);
+			offsetVertex4( width * 0.5, 0.0, -length * 0.5, 0.0, 0.0, 1.414);
+			offsetVertex4(-width * 0.5, 0.0,  length * 0.5, 1.0, 1.0, 1.414);
+			offsetVertex4(-width * 0.5, 0.0, -length * 0.5, 1.0, 0.0, 1.414);
 			EndPrimitive();
 		}
 	#endif
-	
+
 	#ifdef USE_CORNERRECT
-		if (numVertices == uint(2)){ // A quad with chopped off corners
+		if (numVertices == 2u){ // A quad with chopped off corners
 			float csuv = (cs / (length + width))*2.0;
-			offsetVertex4( - width * 0.5 , 0.0,  - length * 0.5 + cs, 0, csuv); // bottom left
-			offsetVertex4( - width * 0.5 , 0.0,  + length * 0.5 - cs, 0, 1.0 - csuv); // top left
-			offsetVertex4( - width * 0.5 + cs, 0.0,  - length * 0.5 , csuv, 0); // bottom left
-			offsetVertex4( - width * 0.5 + cs, 0.0,  + length * 0.5, csuv, 1.0); // top left
-			offsetVertex4( + width * 0.5 - cs, 0.0,  - length * 0.5 , 1.0 - csuv, 0.0); // bottom right
-			offsetVertex4( + width * 0.5 - cs, 0.0,  + length * 0.5 ,1.0 - csuv, 1.0 ); // top right
-			offsetVertex4( + width * 0.5 , 0.0,  - length * 0.5 + cs , 1.0 , csuv ); // bottom right
-			offsetVertex4( + width * 0.5 , 0.0,  + length * 0.5 - cs , 1.0 -csuv , 1.0 ); // top right
+			// FIXME: Need to incorporate corner size to get the best results.
+			// This is a fudge that looks about right for selected buildings, which have corners of 15% for a square.
+			float addRadiusCorrFudge = 1.1;
+			offsetVertex4( - width * 0.5 , 0.0,  - length * 0.5 + cs, 0, csuv, addRadiusCorrFudge); // bottom left
+			offsetVertex4( - width * 0.5 , 0.0,  + length * 0.5 - cs, 0, 1.0 - csuv, addRadiusCorrFudge); // top left
+			offsetVertex4( - width * 0.5 + cs, 0.0,  - length * 0.5 , csuv, 0, addRadiusCorrFudge); // bottom left
+			offsetVertex4( - width * 0.5 + cs, 0.0,  + length * 0.5, csuv, 1.0, addRadiusCorrFudge); // top left
+			offsetVertex4( + width * 0.5 - cs, 0.0,  - length * 0.5 , 1.0 - csuv, 0.0, addRadiusCorrFudge); // bottom right
+			offsetVertex4( + width * 0.5 - cs, 0.0,  + length * 0.5 ,1.0 - csuv, 1.0, addRadiusCorrFudge); // top right
+			offsetVertex4( + width * 0.5 , 0.0,  - length * 0.5 + cs , 1.0 , csuv, addRadiusCorrFudge); // bottom right
+			offsetVertex4( + width * 0.5 , 0.0,  + length * 0.5 - cs , 1.0 -csuv , 1.0, addRadiusCorrFudge); // top right
 			EndPrimitive();
 		}
 	#endif
 	
 	#ifdef USE_CIRCLES
-		if (numVertices > uint(5)) { //A circle with even subdivisions
+		if (numVertices > 5u){ // A circle with even subdivisions
 			numVertices = min(numVertices,62u); // to make sure that we dont emit more than 64 vertices
+			float internalAngle = float(numVertices) * radians(180.0) / float(numVertices);
```

Comment:
Wait, this must be wrong!
Should be (n-2) * 180 / n

I think I got confused because numVertices is pseudo n-2 (when it's 64)

Reply:
Also I think it should do `min(numVertices, 64u)`. IIUC by starting on 1 in the loop we get to 62 vertices?
Or is that's not true I don't know why a hexagon wouldn't emit 8 vertices 😕

Reply:
I'm pretty sure there was a reason I had to limit it to 62 instead of 64. Oh right, cause it actually emits 62+2 vertices for the triangle strip :D But I honestly forgot why. 64 vertices was a hard limit of one set of GPU's anyway.

Reply:
I suspect that you originally had a loop from 0 to vertices/2, but then optimized it by moving the start/end out of the loop? In the current form it works to have a max of 64.
Whether 64 is needed at all for unit markers is an interesting question - ZK used 32 and it was only just about visible at high zoom, with 64 you have to try really hard to see. OTOH, at least on my GPU it makes little-no difference whether I draw 64, 32 or 4, at least when talking about 100 selections.

Reply:
Also on optimization, I wonder how quickly the engine draws the boxes? AIUI it still draws them with 0 opacity when the Lua selection widgets are active.